src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -416,14 +412,29 @@ private function getTagVersions(array $tagsByKey, array &$invalidatedTags = [])
             return $tagVersions;
         }
 
+        $tod = gettimeofday();
```

Comment:
wouldn't it be better to use `mt_rand()`?

Reply:
mt_rand() is a fairly good solution for this problem, it's even simpler than converting microseconds to an integer :) I choose timestamp because of the way of invalidation - incrementing of the value, so it's good to use monotonically growing value wich grows faster than increments. Another consideration was that random numbers may repeat current versions (0, 1, 100000, any of them, I don't know where is the lowest boundary), and the first invalidation will be lost (not more likey than losing any other version, though). Frankly speaking, even with timestamp a version may be overwritten if there are more than one process and there is significant time discrepancy.

So really `mt_rand(0, 2**62-1)` is a neat solution (all 63 bits cannot be used since the counter may overflow large integer to a float).

Reply:
Btw, should I care about 32-bit versions of php?

Reply:
Yes please. a random number would be fine honestly, the monotony doesn't provide anything to me.
About overflows, we should deal with them by looping back to zero (or to PHP_INT_MIN btw if negative numbers are OK)